### PR TITLE
Add sourcehut support, minor cleanups

### PR DESCRIPTION
**Pull request by SuperSandro2000**

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
not sure why this fails.

```
tests/test_git.py .
error: unrecognised flag '--impure'
Try 'nix --help' for more information.
```

File: .envrc [hunk 1]
```diff
@@ -0,0 +1 @@
+eval "$(lorri direnv)"
```
Comment (Mic92) [inline .envrc:1]: Please check if lorri is in path (`if command -v lorry >/dev/null`) and fallback to `use nix` otherwise.

Comment (SuperSandro2000) [reply .envrc:1]:
```suggestion
If type -P lorri; then
  eval "$(lorri direnv)"
else
  nix-shell
fi
```

Comment (Mic92) [reply .envrc:1]: More like:

```suggestion
if has lorri; then
  eval "$(lorri direnv)"
else
  use nix
fi
```

`has()` is a direnv builtin.

**Pull request merged**